cordyceps/src/list.rs

```diff
@@ -1378,6 +1405,64 @@ impl<T: Linked<Links<T>> + ?Sized> DoubleEndedIterator for IterMut<'_, T> {
 
 impl<T: Linked<Links<T>> + ?Sized> iter::FusedIterator for IterMut<'_, T> {}
 
+// === impl RawIter ====
+
+impl<T: Linked<Links<T>> + ?Sized> Iterator for RawIter<'_, T> {
+    type Item = NonNull<T>;
```

Comment:
Hmm, so, in the PR description, you say that this iterates over the `T::Handle` type of the `Linked` implementation for `T`, but here, we actually always return `NonNull<T>`s.

For the record, I think this is more correct. The intent is that `T::Handle` is an _owning_ reference; if `T` isn't something like a static, dropping the `Handle` type should deallocate the `T`, so we would expect it to be implemented for types like `Pin<Box<T>>` and similar. It would definitely be wrong for the raw iterator to yield `Box<T>`s containing elements that are still in the list, as the iterator items would drop them. So, I think the use of `NonNull` here is correct.

Reply:
I think the issue pointed at handle, in the pr I mention that nonnull is probably the practical right choice.

Reply:
Oh, I read the wrong thing, whoops.